compiler-core/src/javascript/expression.rs

```diff
@@ -593,7 +593,10 @@ impl<'module> Generator<'module> {
                     }
                     // Create an assignment for each variable created by the function arguments
                     if let Some(name) = argument {
-                        docs.push(Document::String(format!("{} = ", name)));
+                        // TODO escape here
+                        // docs.push(Document::String(format!("{}!! = ", name)));
+                        docs.push(maybe_escape_identifier(name));
```

Comment:
This seemed like a simple fix. But I got many lifetime errors

```
error[E0495]: cannot infer an appropriate lifetime due to conflicting requirements
   --> compiler-core/src/javascript/expression.rs:588:50
    |
588 |                     .zip(self.function_arguments.clone())
    |                                                  ^^^^^
    |
note: first, the lifetime cannot outlive the lifetime `'module` as defined on the impl at 34:6...
   --> compiler-core/src/javascript/expression.rs:34:6
    |
34  | impl<'module> Generator<'module> {
    |      ^^^^^^^
note: ...so that the types are compatible
   --> compiler-core/src/javascript/expression.rs:588:50
    |
588 |                     .zip(self.function_arguments.clone())
    |                                                  ^^^^^
    = note: expected `&Vec<std::option::Option<&str>>`
               found `&Vec<std::option::Option<&'module str>>`
note: but, the lifetime must be valid for the lifetime `'a` as defined on the method body at 536:27...
   --> compiler-core/src/javascript/expression.rs:536:27
    |
536 |     fn call_with_doc_args<'a>(
    |                           ^^
note: ...so that the expression is assignable
   --> compiler-core/src/javascript/expression.rs:607:17
    |
607 |                 Ok(docs.to_doc())
    |                 ^^^^^^^^^^^^^^^^^
    = note: expected `std::result::Result<pretty::Document<'a>, _>`
               found `std::result::Result<pretty::Document<'_>, _>`

```

Reply:
oh how strange. I'll investigate

Reply:
This is a fiddly one to do with lifettimes so I've just duplicated some code and cloned it.